integration_test/pathfilter_integration_test.go

```diff
@@ -0,0 +1,112 @@
+package integrationtest
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openebs/node-disk-manager/integration_test/ndm_util"
+	"time"
+)
+
+var _ = Describe("Path filter integration Test", func() {
+
+	var configMapPatch ndmutil.ConfigMapPatch
+	var err error
+
+	When("Path filter is disabled", func() {
+		err = ndmutil.InitEnvironment()
```

Comment:
I don't think it will work. `BeforeEach` is executed just before the `It` block. Moving the `InitEnvironment` to `BeforeEach` will thus cause panic. Because the rest of the code in `When` block can only be executed after Initializing the environment

Reply:
Right @akhilerm . What I was trying to convey was; can we design these tests in such a way so that we would be able to use `BeforeEach`. Because it looks redundent to me. But still it would be your call, if every particular test will need to setup the environment again, its better to use `BeforeEach`.

Reply:
Just a Suggestion here, may be you could use`Specify` instead of `When`. Or else if you are OK with this I won't mind.

Reply:
Do you want to continue with this very approach @akhilerm ?

Reply:
@a4abhishek I think we can go with this way of initialization. I can't think of another way to do the initialization.

Reply:
OK